cpp/open3d/t/pipelines/kernel/ComputePosePointToPlane.cpp

```diff
@@ -34,43 +34,50 @@ namespace pipelines {
 namespace kernel {
 
 core::Tensor ComputePosePointToPlane(
-        const core::Tensor &source_points_indexed,
-        const core::Tensor &target_points_indexed,
-        const core::Tensor &target_normals_indexed) {
+        const core::Tensor &source_points,
+        const core::Tensor &target_points,
+        const core::Tensor &target_normals,
+        const pipelines::registration::CorrespondenceSet &corres) {
     // Get dtype and device.
-    core::Dtype dtype = source_points_indexed.GetDtype();
-    core::Device device = source_points_indexed.GetDevice();
+    core::Dtype dtype = source_points.GetDtype();
+    core::Device device = source_points.GetDevice();
 
     // Checks.
     // TODO: These checks are redundant, so provide a environment
     // variable based method to skip these redundant tests.
-    target_points_indexed.AssertDtype(dtype);
-    target_normals_indexed.AssertDtype(dtype);
-    target_points_indexed.AssertDevice(device);
-    target_normals_indexed.AssertDevice(device);
+    target_points.AssertDtype(dtype);
+    target_normals.AssertDtype(dtype);
+    target_points.AssertDevice(device);
+    target_normals.AssertDevice(device);
 
     // Pose {6,} tensor [ouput].
     core::Tensor pose = core::Tensor::Empty({6}, dtype, device);
-
     // Number of correspondences.
-    int n = source_points_indexed.GetShape()[0];
+    int n = corres.first.GetShape()[0];
 
     // Pointer to point cloud data - indexed according to correspondences.
-    const float *src_pcd_ptr = static_cast<const float *>(
-            source_points_indexed.Contiguous().GetDataPtr());
-    const float *tar_pcd_ptr = static_cast<const float *>(
-            target_points_indexed.Contiguous().GetDataPtr());
+    const float *src_pcd_ptr =
+            static_cast<const float *>(source_points.Contiguous().GetDataPtr());
```

Comment:
We need to do it in two steps:
```cpp
Tensor source_points_contiguous = source_points.Contiguous();
const float *src_pcd_ptr = static_cast<const float *>(source_points_contiguous.GetDataPtr());
```

Otherwise, if `source_points` is not yet contiguous, `source_points.Contiguous()` will go out-of-scope such that `source_points.Contiguous().GetDataPtr()` is not valid any longer. 

We will need to do the same for the rest of the `Contiguous()` calls as well.